docs/testing/continuous-integration.md

```diff
@@ -52,16 +49,40 @@ uses those SSH keys for authentication.
 
 The main CircleCI configuration file is
 [./circleci/config.yml](https://github.com/zulip/zulip/blob/master/.circleci/config.yml).
-We run multiple jobs during a CircleCI build to test different
-supported platforms. They are currently:
+We run multiple jobs during a CircleCI build to test both development and production environment
+in different supported platforms. They are currently:
 
+For development -
 * bionic-backend-python3.6
+* focal-backend-python3.8
 
 Each runs the Zulip backend test suites, using the indicated
 platform/OS and Python version.  As suggested by the names, only one
 suite runs the frontend test suites, since those are not
 platform-dependent.
 
+For production -
+* bionic-production-build-python3.6
+* bionic-production-install-python3.6
+
+Currently we test our production suite on Ubuntu Bionic 18.04.
+To carry out this process we use the two jobs stated above. Job
+bionic-production-build-python3.6 builts the release tarball and
+once this job finishes the tarball is then used by our downstream
+job bionic-production-install-python3.6 for installation of the production suite.
+
+We use a few helper scripts to setup the production suites:
+
+* `tools/ci/setup-production`: This script builts the release
+   tarball for the production suite.
+* `tools/ci/production-extract-tarball`: Extract the tarball for
+  production suites.
+* `tools/ci/production`: It installs and run production tests.
+
+Since we dropped the support for Ubuntu Xenial 16.04 it is important that we
+test our upgradation process. So the job xenial-legacy is to
+test provision and upgradation on Ubuntu 16.04.
```

Comment:
I decided to remove this last block about helper scripts, and just have explanation above for how to find these.